src/Symfony/Component/DependencyInjection/Tests/Compiler/IntegrationTest.php

```diff
@@ -234,6 +239,76 @@ public function getYamlCompileTests()
+    {
+        $includeFile = realpath(__DIR__.'/../Fixtures/includes/bar.php');
+
+        $container = new ContainerBuilder();
+        $container->register('bar_tag_service', BarTagClass::class)
+            ->setFile($includeFile)
+            ->setPublic(true)
+            ->addTag('foo_bar', ['foo' => 'foo'])
+        ;
+        $container->register('foo_tag_service', FooTagClass::class)
+            ->setFile($includeFile)
+            ->setPublic(true)
+            ->addTag('foo_bar', ['foo' => 'bar'])
+        ;
+        $container->register('foobar_tagged_service', FooBarTaggedClass::class)
+            ->setFile($includeFile)
+            ->addArgument(new TaggedIteratorArgument('foo_bar', 'foo'))
+            ->setPublic(true)
+        ;
+
+        $container->compile();
+        $this->assertNotNull($container->getDefinition('bar_tag_service'));
+        $this->assertNotNull($container->getDefinition('foo_tag_service'));
+        $this->assertNotNull($container->getDefinition('foobar_tagged_service'));
+
+        $s = $container->get('foobar_tagged_service');
+        $this->assertTrue($s instanceof FooBarTaggedClass, 'The service foobar_tagged_service should be an instance of Bar\FooBarTaggedClass');
+
+        $param =  iterator_to_array($s->getParam()->getIterator());
+        $this->assertCount(2, $param);
+        $this->assertArrayHasKey('foo', $param);
+        $this->assertArrayHasKey('bar', $param);
+    }
+
+    public function testTaggedServiceWithIndexAttributeAndDefaultMethod()
+    {
+        $includeFile = realpath(__DIR__.'/../Fixtures/includes/bar.php');
+
+        $container = new ContainerBuilder();
+        $container->register('bar_tag_service', BarTagClass::class)
+            ->setFile($includeFile)
+            ->setPublic(true)
+            ->addTag('foo_bar', ['foo' => 'foo'])
+        ;
+        $container->register('foo_tag_service', FooTagClass::class)
+            ->setFile($includeFile)
+            ->setPublic(true)
+            ->addTag('foo_bar', ['foo' => 'bar'])
+        ;
+        $container->register('foobar_tagged_service', FooBarTaggedClass::class)
+            ->setFile($includeFile)
+            ->addArgument(new TaggedIteratorArgument('foo_bar', 'foo', 'getFooBar'))
+            ->setPublic(true)
+        ;
+
+        $container->compile();
+        $this->assertNotNull($container->getDefinition('bar_tag_service'));
+        $this->assertNotNull($container->getDefinition('foo_tag_service'));
+        $this->assertNotNull($container->getDefinition('foobar_tagged_service'));
+
+        $s = $container->get('foobar_tagged_service');
+        $this->assertTrue($s instanceof FooBarTaggedClass, 'The service foobar_tagged_service should be an instance of Bar\FooBarTaggedClass');
+
+        $param =  iterator_to_array($s->getParam()->getIterator());
+        $this->assertCount(2, $param);
+        $this->assertArrayHasKey('foo', $param);
+        $this->assertArrayHasKey('bar', $param);
```

Comment:
we should verify the values also, not only the keys

Reply:
Values are service objects, I can test with instance of :thinking:

Reply:
or with `$container->get('..')`, isn't it?

Reply:
yes indead.